torch/testing/_internal/common_methods_invocations.py

```diff
@@ -8792,6 +8793,20 @@ def reference_inputs_flatten(op, device, dtype, requires_grad, **kwargs):
         yield SampleInput(make_arg(shape, noncontiguous=True).transpose(0, -1), args=(start, end,))
         yield SampleInput(make_arg(shape).transpose(0, -1), args=(start, end,))
 
+def sample_inputs_unflatten(op_info, device, dtype, requires_grad, **kwargs):
+    # in_shape, dim, sizes
+    args = (((8,), 0, (8,)),
+            ((8,), 0, (4, 2)),
+            ((8,), -1, (2, 2, 2)),
+            ((3, 6, 2), 1, (2, 3)),
+            ((3, 6, 2), -2, (2, 3)),
+            ((3, 2, 12), 2, (3, 2, 2))
```

Comment:
Unflatten (similar to view) accepts negative dimensions for one of the desired dims (e.g. `a.unflatten(2, (2,-1))), we should test that too, as well as 0 dims - because these things are easy to miss when implementing a reference.

Reply:
Thanks, I added a couple more test cases that should cover that.